lib/authorization.js

```diff
@@ -3,14 +3,22 @@ const errors = require('./errors'), rparams = require('./param'), sender = requi
   constants = require('./constants').get(),
   csauth = constants.header.AUTH,
   fields = {}, params = {};
+// initialize [TODO: overridable] global default
+var unauthenticatedcode = constants.httpcodes.UNAUTHENTICATED;
 
-const checkLoggedIn = (req, res, next) => {
+const checkLoggedIn = (req, res, next, unauthenticatedResponse = {code:unauthenticatedcode}) => {
     // If our user is authenticated
     // then everything is fine :)
     let logged = true;
     rparams(req);
     if (!req[csauth]) {
-      sender(res,401,errors.unauthenticated());
+      unauthenticatedResponse = unauthenticatedResponse || {code:unauthenticatedcode};
+      unauthenticatedResponse.code = unauthenticatedResponse.code || unauthenticatedcode; 
+      unauthenticatedResponse.location = unauthenticatedResponse.location || null; 
+      if (unauthenticatedResponse.location != null) {
+        res.header("location", unauthenticatedResponse.location);
+      }
+      sender(res,unauthenticatedResponse.code,errors.unauthenticated(),unauthenticatedResponse);
```

Comment:
How does this work? `sender()` only accepts 3 args: `res, code, body`. Here you send 4?